client/core/trade.go

```diff
@@ -1796,79 +1813,115 @@ func (t *trackedTrade) refundMatches(matches []*matchTracker) (uint64, error) {
+		t.mtx.Lock()
+		auth := &match.MetaMatch.MetaData.Proof.Auth
+		auth.AuditStamp, auth.AuditSig = audit.Time, audit.Sig
+		err = t.db.UpdateMatch(&match.MetaMatch)
+		t.mtx.Unlock()
+		if err != nil {
+			t.dc.log.Errorf("Error updating database for match %v: %v", match.id, err)
+		}
 
-	err = t.db.UpdateMatch(&match.MetaMatch)
-	if err != nil {
-		return errs.add("error updating database: %v", err)
-	}
+		// Respond to DEX, but this is not consequential.
+		err = t.dc.ack(msgID, match.id, audit)
+		if err != nil {
+			t.dc.log.Debugf("Error acknowledging audit to server (not necessarily an error): %v", err)
+			// The server's response timeout may have just passed, but we got
+			// what we needed to do our swap or redeem if the match is still
+			// live, so do not log this as an error.
+		}
+	}()
 
-	// Respond to DEX.
-	err = t.dc.ack(msgID, match.id, audit)
-	if err != nil {
-		return errs.add("Audit error: %v", err)
-	}
 	return nil
 }
 
 // auditContract audits the contract for the match and relevant MatchProof
-// fields are set. The changes are not saved to the database. auditContract must
-// be called with the trackedTrade.mtx locked.
+// fields are set. The changes are not saved to the database. This may block for
+// a long period, and should be run in a goroutine. The trackedTrade mtx must
+// NOT be locked. The match is updated in the DB if the audit succeeds.
 func (t *trackedTrade) auditContract(match *matchTracker, coinID []byte, contract []byte) error {
 	// Get the asset.AuditInfo from the ExchangeWallet. Handle network latency.
 	// The coin waiter will run once every recheckInterval until successful or
-	// until expiration after broadcast timeout. The client is asked by the
-	// server to audit a contract transaction, and they have until broadcast
-	// timeout to do it before they get penalized and the match revoked. Thus,
-	// there is no reason to give up on the request sooner since the server will
-	// not ask again.
+	// until the match is revoked. The client is asked by the server to audit a
+	// contract transaction, and they have until broadcast timeout to do it
+	// before they get penalized and the match revoked. Thus, there is no reason
+	// to give up on the request sooner since the server will not ask again and
+	// the client will not solicit the counterparty contract data again except
+	// on reconnect.
 	errChan := make(chan error, 1)
 	var auditInfo asset.AuditInfo
+	var tries int
+	contractID, contractSymb := coinIDString(t.wallets.toAsset.ID, coinID), t.wallets.toAsset.Symbol
 	t.latencyQ.Wait(&wait.Waiter{
-		Expiration: time.Now().Add(t.broadcastTimeout()),
+		Expiration: time.Now().Add(24 * time.Hour), // effectively forever
 		TryFunc: func() bool {
 			var err error
 			auditInfo, err = t.wallets.toWallet.AuditContract(coinID, contract)
-			if err != nil {
-				if errors.Is(err, asset.CoinNotFoundError) {
-					return wait.TryAgain
-				}
-				errChan <- err
+			if err == nil {
```

Comment:
@buck54321 The hack I was using to simulate slow wallet AuditContract discovery of the coin is the following right after `AuditContract`:

```go
if tries < 13 {
	err = asset.CoinNotFoundError
}
```